### PR TITLE
add hyperparameter functionality

### PR DESCRIPTION
**Pull request by aflower15**
closes issues #41 and #42

### CODE REVIEW

File: src/pipelines/master_pipeline.py [hunk 1]
```diff
@@ -90,8 +85,119 @@ def parse_args(given_args=None):
...
+    if log_reg: algorithms['log_reg'] = log_reg
+    if svm: algorithms['svm'] = svm
+    
+    resampling = None
+    
+    if RESAMPLING:
+        resampling = dict()
+        if NOVEL_RATIO: resampling['novelToNotNovelRatio'] = NOVEL_RATIO
+        if OVERSAMPLING: resampling['over'] = OVERSAMPLING
+        if REPLACEMENT: resampling['replacement'] = replacement
```
Comment (bethke) [inline src/pipelines/master_pipeline.py]: replacement should be all caps to match above REPLACEMENT

File: src/pipelines/master_pipeline.py [hunk 2]
```diff
@@ -7,47 +7,42 @@
 import sys
 import argparse
 from collections import namedtuple
-from src.pipelines import parse_json, preprocess, observations, features_and_labels, log_reg, svm
+from src.pipelines import parse_json, preprocess, data_gen, log_reg, svm, predict
```
Comment (bethke) [inline src/pipelines/master_pipeline.py]: Looks like this commit may be missing data_gen and predict...

File: src/pipelines/log_reg.py [hunk 3]
```diff
@@ -33,26 +32,22 @@ def run_model(train_data, train_labels, test_data, test_labels, log_penalty='l2'
...
     print("running logistic regression...",file=sys.stderr)
 
-    predicted_labels, perform_results = run_model(train_data, train_target, test_data, test_target, **args_dict)
+    logreg = run_model(train_data, train_target, **args_dict)
 
-    return predicted_labels, perform_results
+    return logreg
 
 if __name__ == '__main__':
     if len(sys.argv) < 5:
```
Comment (dave-lab41) [inline src/pipelines/log_reg.py]: The __main__ needs be updated to reflect that we no longer expect sys.argv to have a length of 5 and the usage note doesn't need test_data or test_target anymore

File: src/pipelines/svm.py [hunk 4]
```diff
@@ -30,27 +29,23 @@ def run_model(train_data, train_labels, test_data, test_labels, svm_C=1.0, svm_k
...
     print("running support vector machine...",file=sys.stderr)
 
-    predicted_labels, perform_results = run_model(train_data, train_target, test_data, test_target, **args_dict)
+    svm = run_model(train_data, train_target, **args_dict)
 
-    return predicted_labels, perform_results
+    return svm
 
 if __name__ == '__main__':
     if len(sys.argv) < 5:
```
Comment (dave-lab41) [inline src/pipelines/svm.py]: The usage section needs be updated to reflect that we no longer expect sys.argv to have a length of 5 and the print statement should remove test_data and test_labels

**Pull request merged**